src/components/CountryDetails.js

```diff
@@ -0,0 +1,46 @@
+import { useParams } from "react-router-dom";
+import { useState, useEffect } from "react";
+import axios from "axios";
+
+function CountryDetails(props){
+   
+    const{id} = useParams();
+    const[details, setDetails] = useState("")
+    let bigId = id.toUpperCase()
+    console.log(bigId)
+
+// useEffect(()=>{
+    
+    
+//     axios.get(`https://ih-countries-api.herokuapp.com/countries/${bigId}`)
+//     .then(response=>{
+//         setDetails(response)
+//         console.log(response)
+//     })
+//     .catch()
+// })
+
+const filterArray = ()=>{
+   
+        props.countriesArr.filter((country)=>{
+
+            {(country.alpha3Code === id) ?  
+            (setDetails(country.alpha3Code))
+            : (console.log("error"))}
+            
+                })
+ 
+  
+}
```

Comment:
Also ich hab deinen Code etwas verändert und erst mal nur den Namen anzeigen lassen:
```
import { useParams } from "react-router-dom";
import { useState, useEffect } from "react";

function CountryDetails(props){
    const {id} = useParams();
    const [details, setDetails] = useState("")

    //du hattest mit filter() angefangen (gibt ein Array zurück), aber ich würde dir hier find() empfehlen (gibt nur den Wert zurück)
    const getDetails = ()=> {
        let country = props.countriesArr.find((country)=>{
            return country.alpha3Code === id
         })
         setDetails(country)
    }
  
    useEffect(()=> {
        getDetails()
    }, [])
  

if (details == ""){
   return <p>Loading...</p>
}

return(

   <h1>{details.name.common}</h1>
)


}

export default CountryDetails;
```

Reply:
Ich habe useEffect benutzt, damit die Funktion aufgerufen wird, sobald die Seite geladen wird. Also die country die zu dem alpha3Code passt dann in details gespeichert wird. Die if Bedingung for Loading mache ich um einen error zu vermeiden. Also solange details noch ein empty string ist zeige Loading und danach zeige mir den namen des Landes.

Lass mich wissen, falls du dazu noch Fragen hast.